### DISCUSSION TITLE
Commenting for Solution Implementation

### DISCUSSION THREAD

**Discussion by oraqlle** [PlummersSoftwareLLC/Primes#703] [Ideas]
I'd imagine that amongst Dave Plummer's YouTube viewer base there is a wide variety of beginners, novices and experts and while many higher end developers may want to flex their optimising prowess for some beginners (such as myself who hasn't yet taught myself algorithms yet so please excuse my possible ignorance on topics), having comments around the various solutions and implantations for every language would go a long way in explaining how the code works and why an author has chosen to implement their algorithm a specific way. 
Even if other contributors who didn't write the code itself but understand how and why it is implemented, I think it would be greatly appreciated amongst the community. With so many languages throughout this repo, it would be cool to be able to navigate to a new language solution and see not only how the solution works but how that language works as well. 
I know a few languages/implementations have already got commenting which is awesome but I think it should be a goal of the community to try and have comments around every solution. This could also clear up obscurity and misrepresentation when the language gets featured in a video when Dave is addressing the solution for the language. 
I know this might not be high on the priority list but maybe something to consider as loosely required for a solution to be valid. 
Just food for though.

**Comment by rbergen** [comment]
I do think source code comments can be very helpful towards making source code readable by others, which is why I have added them to some of the solutions I submitted (the Assembly solutions being the most striking examples). 
That said, the discussion of what is an _appropriate_ amount of in-source documentation is a very subjective one. Too much documentation can actually obfuscate the meaning of the actual source code, and source code really can be self-documenting. 

Currently, we accept source code comments and require the addition of a README.md to every solution. The latter is quite regularly used by contributors to explain the (algorithmic) approach they have taken to perform the sieve calculations within their solution. In those cases we (the maintainers) need help with understanding source code that's contributed while reviewing, we also do ask for it; none of us is an expert in all 75 or so languages we currently have in the repo.
On top of that, in the web application we provide to view benchmark reports (PrimeView), we provide links to webpages concerning the languages included in the reports.

Without ruling out a future development in this area, I currently don't see a reason to include comments to the contribution requirements in any way. If we were to consider doing so, the concept of "loosely requiring" it would really need a more objective/formal definition. That would be an interesting challenge, in itself.
